buffer/src/main/java/io/netty/buffer/AdaptivePoolingAllocator.java

```diff
@@ -739,13 +751,15 @@ private void deallocate() {
                 mag.usedMemory.getAndAdd(-capacity());
                 delegate.release();
             } else {
-                updater.resetRefCnt(this);
                 delegate.setIndex(0, 0);
                 allocatedBytes = 0;
-                if (!mag.trySetNextInLine(this)) {
-                    if (!parent.offerToQueue(this)) {
-                        // The central queue is full. Drop the memory with the original Drop instance.
-                        delegate.release();
+
+                // Create a new chunk which holds the delegate that can be reused.
+                Chunk chunk = new Chunk(this);
+                if (!mag.trySetNextInLine(chunk)) {
+                    if (!parent.offerToQueue(chunk)) {
+                        // The central queue is full. Drop the chunk which will also release the delegate.
+                        chunk.release();
```

Comment:
Might this not be recursive and come back to here, but in the chunk we just created?

Reply:
Hmm yeah you are right ... maybe let's just do what I had in the other PR before